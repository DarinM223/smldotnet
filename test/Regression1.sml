structure Regression1 = struct
  fun main () =
    let
      val f = ref ()
      val () = f := ()
    in
      ()
    end
end