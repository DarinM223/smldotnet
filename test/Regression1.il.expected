.assembly extern mscorlib {
  .publickeytoken = (B7 7A 5C 56 19 34 E0 89 )
  .ver 4:0:0:0
}
.assembly extern Accessibility {
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 2:0:0:0
}
.assembly extern System.Configuration.Install {
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 2:0:0:0
}
.assembly extern System.Data {
  .publickeytoken = (B7 7A 5C 56 19 34 E0 89 )
  .ver 2:0:0:0
}
.assembly extern System.DirectoryServices {
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 2:0:0:0
}
.assembly extern System {
  .publickeytoken = (B7 7A 5C 56 19 34 E0 89 )
  .ver 2:0:0:0
}
.assembly extern System.Drawing {
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 2:0:0:0
}
.assembly extern System.EnterpriseServices {
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 2:0:0:0
}
.assembly extern System.Management {
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 2:0:0:0
}
.assembly extern System.Messaging {
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 2:0:0:0
}
.assembly extern System.Runtime.Remoting {
  .publickeytoken = (B7 7A 5C 56 19 34 E0 89 )
  .ver 2:0:0:0
}
.assembly extern System.Runtime.Serialization.Formatters.Soap {
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 2:0:0:0
}
.assembly extern System.Security {
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 2:0:0:0
}
.assembly extern System.ServiceProcess {
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )
  .ver 2:0:0:0
}
.assembly extern Mono.Unix {
  .publickeytoken = (CC 7B 13 FF CD 2D DD 51 )
  .ver 7:1:0:0
}
.assembly 'Regression1'
{ .hash algorithm 0x00008004 }
.module 'Regression1.exe'

.namespace '$' {
.class  private abstract 'Class_a' extends ['mscorlib']'System'.'Exception'{

.field private initonly string 's'
.method public instance void '.ctor'(string 's') cil {
.maxstack 2
.zeroinit
  ldarg.0
  ldarg.1
  call instance instance void ['mscorlib']'System'.'Exception'::'.ctor'(string)
  ldarg.0
  ldarg.1
  stfld string '$'.'Class_a'::'s'
  ret
{.locals ()}
}
.method public virtual instance string 'ToString'() cil {
.maxstack 3
.zeroinit
  ldarg.0
  callvirt instance string '$'.'Class_a'::'ExnMessage'()
  stloc.0
  ldsfld string '$'.'Globals'::'$l'
  stloc.1
  ldloc.1
  ldnull
  ceq
  stloc.2
  ldloc.2
  brtrue L_1
  ldloc.0
  ldloc.1
  call string ['mscorlib']'System'.'String'::'Concat'(string,string)
  stloc.0
  ldloc.0
  ret
L_1:
  ldloc.0
  ret
{.locals ([0] string /*cryt 'PrimUtils_.prefix'*/ ,
         [1] string /*cryr*/ ,
         [2] bool /*csaa*/ )}
}
.method public virtual instance string 'ExnMessage'() cil {
.maxstack 2
.zeroinit
  ldarg.0
  callvirt instance string '$'.'Class_a'::'ExnName'()
  stloc.0
  ldloc.0
  ldstr ": "
  call string ['mscorlib']'System'.'String'::'Concat'(string,string)
  stloc.0
  ldarg.0
  callvirt instance string ['mscorlib']'System'.'Exception'::'get_Message'()
  stloc.1
  ldloc.0
  ldloc.1
  call string ['mscorlib']'System'.'String'::'Concat'(string,string)
  stloc.0
  ldloc.0
  ret
{.locals ([0] string /*cryz cryx cryw*/ ,
         [1] string /*cryy*/ )}
}
.method public virtual instance string 'ExnName'() cil {
.maxstack 1
.zeroinit
  ldstr "SML exception"
  ret
{.locals ()}
}

}
}

.class  public sealed 'Regression1' extends ['mscorlib']'System'.'Object'{

.method public static void '.cctor'() cil {
.maxstack 0
.zeroinit
  call void '$'.'Globals'::'$'()
  ret
{.locals ()}
}
.method public static void 'main'() cil {
.maxstack 0
.zeroinit
.entrypoint
  ret
{.locals ()}
}

}

.namespace '$' {
.class  private 'Fun' extends ['mscorlib']'System'.'Object'{

.method assembly instance void '.ctor'() cil {
.maxstack 2
.zeroinit
  ldarg.0
  call instance void ['mscorlib']'System'.'Object'::'.ctor'()
  ret
{.locals ()}
}

}
}

.namespace '!'.'General' {
.class  private sealed 'Fail' extends '$'.'Class_a'{

.method assembly instance void '.ctor'(string '') cil {
.maxstack 3
.zeroinit
  ldarg.0
  ldarg.1
  call instance void '$'.'Class_a'::'.ctor'(string)
  ret
{.locals ()}
}
.method public virtual instance string 'ExnName'() cil {
.maxstack 3
.zeroinit
  ldstr "General.Fail"
  ret
{.locals ()}
}

}
}

.namespace '$' {
.class  private 'Globals' extends ['mscorlib']'System'.'Object'{

.field static assembly string '$l'
.field static assembly int32 '$g'
.method static assembly void '.cctor'() cil {
.maxstack 3
.zeroinit
  ldc.i4.s 11
  conv.u4
  stloc.0
  ldc.i4.1
  ldloc.0
  shl
  stloc.1
  ldloc.1
  ldc.i4.1
  sub.ovf
  pop
  ldc.i4.s 11
  ldc.i4.1
  sub.ovf
  stloc.1
  ldloc.1
  conv.u4
  stloc.0
  ldc.i4.1
  ldloc.0
  shl
  stloc.1
  ldloc.1
  ldc.i4.1
  sub.ovf
  stloc.1
  ldloc.1
  ldc.i4.s 52
  add.ovf
  stloc.2
  ldloc.2
  ldc.i4.1
  sub.ovf
  pop
  ldc.i8 0x0000000000000040
  conv.ovf.i4
  stloc.2
  ldloc.2
  ldc.i4.1
  sub.ovf
  stloc.2
  ldloc.2
  ldloc.1
  add.ovf
  pop
  call string ['mscorlib']'System'.'Environment'::'get_NewLine'()
  stloc.3
  ldloc.3
  ldnull
  ceq
  stloc.s 4
  ldloc.s 4
  brtrue L_5
  ldloc.3
  ldc.i4.s -2147483648
  callvirt instance char ['mscorlib']'System'.'String'::'get_Chars'(int32)
  stloc.s 5
  ldloc.3
  callvirt instance int32 ['mscorlib']'System'.'String'::'get_Length'()
  stloc.1
  ldloc.1
  dup
  ldc.i4.1
  bne.un L_3
  pop
  ldloc.s 5
  ldc.i4 10
  ceq
  stloc.s 4
  ldloc.s 4
  brtrue L_2
  br _crzq
L_2:
  br _crzq
L_3:
  ldc.i4.2
  bne.un L_4
  ldloc.3
  ldc.i4.1
  callvirt instance char ['mscorlib']'System'.'String'::'get_Chars'(int32)
  stloc.s 6
  br _crzq
L_4:
  ldstr " at TextIO:125.26-79"
  stsfld string '$'.'Globals'::'$l'
  ldstr "Unexpected line separator length"
  newobj instance void '!'.'General'.'Fail'::.ctor(string)
  throw
_crzq:
  call class ['mscorlib']'System'.'IO'.'TextReader' ['mscorlib']'System'.'Console'::'get_In'()
  pop
  call class ['mscorlib']'System'.'IO'.'TextWriter' ['mscorlib']'System'.'Console'::'get_Out'()
  pop
  call class ['mscorlib']'System'.'IO'.'TextWriter' ['mscorlib']'System'.'Console'::'get_Error'()
  pop
  ret
L_5:
  ldstr " at TextIO:118.20-68"
  stsfld string '$'.'Globals'::'$l'
  ldstr "No line separator available"
  newobj instance void '!'.'General'.'Fail'::.ctor(string)
  throw
{.locals ([0] unsigned int32 /*crzf crzc*/ ,
         [1] int32 /*crzp 'Real.eoffset' crzg crze crzd*/ ,
         [2] int32 /*crzk 'Real.large_precision' 'Real.meoffset'*/ ,
         [3] string /*crzl*/ ,
         [4] bool /*crzw crzz*/ ,
         [5] char /*'TextIO.c'*/ ,
         [6] char /*crzx*/ )}
}
.method static assembly void '$'() cil {
.maxstack 0
.zeroinit
  ret
{.locals ()}
}
.method static assembly void '.GCSafePoint'() managed noinlining {
.maxstack 0
.zeroinit
  ret
{.locals ()}
}

}
}
